### PR TITLE
Please Robot Overlords

### PR DESCRIPTION
**Pull request by nmking22**
### Description
Fixed all Rubocop infractions by:
* Extracting helper methods on lengthy methods
* Changing improper method names
* Adding dependent destroys to has_many relationships

Added additional logic to solve edge cases of users trying to add themselves or existing friends as new friends
Adapted unauthorized user view displays to all be consistent

### Checklist
- [x] All tests passing
- [x] Sad paths fully tested (or needed tests added to project board)
- [x] All co-authors added to commits

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -2,15 +2,12 @@ class SessionsController < ApplicationController
   def create
     user = User.find_by(email: params[:email])
     if !user
-      flash[:error] = 'Email and/or password is incorrect'
-      redirect_to '/'
+      redirect_to welcome_path, notice: 'Email and/or password is incorrect'
```
Comment (ShaunDaneJames) [inline app/controllers/sessions_controller.rb:5]: Neat way to handle the flash message!

File: app/facades/movie_facade.rb [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
     movies = []
-    page = 1
     until movies.count >= 40
-      movies_data = MovieService.find_by_title(title, page)
+      movies_data = MovieService.search_by_title(title, page ||= 1)
       movies_data.each do |search_data|
         movies << MoviePoro.new(search_data)
       end
-      # leave comment here explaining the usage of the number 20
+      # the magic number 20 is used here, because that is the default yield per page from this API
```
Comment (ShaunDaneJames) [inline app/facades/movie_facade.rb:9]: Could we have stored this magic number in a variable which could have explained its purpose?

**Review by ShaunDaneJames — COMMENTED** [review]
Great job Nick!

File: app/facades/movie_facade.rb (hunk 2 above)
Comment (nmking22) [reply app/facades/movie_facade.rb:9]: That's a great observation. I think you're 100% right here. This could be a good opportunity for something to bring up in the eval for the 'Team can identify areas where code can be refactored and how they may have created technical debt' requirement in the rubric.

**Review by GarrettCottrell — APPROVED** [review]
